File: apps/windows/src/Services/SyncHistoryStore.cs
// Raw read/write for sync_history.json — own file, separate from
// settings.json (which is load-merge-saved by several unrelated services;
// this is append-mostly operational data, not a preference).
using System.IO;
using System.Text.Json;
using LexiCall.Desktop.Models;

namespace LexiCall.Desktop.Services;

internal static class SyncHistoryStore
{
    public const int MaxEntries = 200;

    private static readonly string HistoryFilePath = Path.Combine(
        Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
        "LexiCall",
        "sync_history.json");

    public static List<SyncHistoryEntry> Load()
    {
        try
        {
            if (!File.Exists(HistoryFilePath))
            {
                return [];
            }

            return JsonSerializer.Deserialize<List<SyncHistoryEntry>>(File.ReadAllText(HistoryFilePath))
                ?? [];
        }
        catch (Exception exception) when (exception is IOException or JsonException)
        {
            // A corrupted history file must never block startup.
            return [];
        }
    }

    // Entries are expected newest-first; trims to MaxEntries here so every
    // caller gets the cap for free.
    public static void Save(IReadOnlyList<SyncHistoryEntry> entries)
    {
        try
        {
            var toSave = entries.Count > MaxEntries ? entries.Take(MaxEntries).ToList() : entries;
            Directory.CreateDirectory(Path.GetDirectoryName(HistoryFilePath)!);
            File.WriteAllText(HistoryFilePath, JsonSerializer.Serialize(toSave));
        }
        catch (IOException)
        {
            // Not saved: harmless, history stays active in memory for the session.
        }
    }
}

File: apps/android/src/services/syncHistoryStore.js
// Read/write for sync_history.json — its own file, kept apart from
// settings.json: append-mostly operational data, not a preference.
import { File, Paths } from 'expo-file-system';

export const MAX_HISTORY_ENTRIES = 200;

const HISTORY_FILE = 'sync_history.json';

function historyFile() {
  return new File(Paths.document, HISTORY_FILE);
}

export async function loadSyncHistory() {
  const file = historyFile();

  if (!file.exists) {
    return [];
  }

  try {
    const parsed = JSON.parse(await file.text());
    return Array.isArray(parsed) ? parsed : [];
  } catch {
    // A corrupted history file must never block startup.
    return [];
  }
}

// Entries come in newest-first; trimming here gives every caller the cap.
export function saveSyncHistory(entries) {
  const file = historyFile();

  if (!file.exists) {
    file.create({ intermediates: true });
  }
  file.write(JSON.stringify(entries.slice(0, MAX_HISTORY_ENTRIES)));
}
